### PR TITLE
Fix #3203 App Crash when rotating the Peer Review Screen

### LINKED ISSUE #3203: Review: App crash when rotating the Peer Review screen
**Summary:** 

Summarize your issue in one sentence (what goes wrong, what did you expect to happen)

App is getting crash when I rotate the screen for "Is this correctly categorized?" title.

**Steps to reproduce:** 

How can we reproduce the issue? 
What did you expect the app to do, and what did you see instead?

Steps to reproduce

1. Go to the Navigation Drawer and click on "Review".
2.  Click "seems fine" 2 times for the image.
3. Now "Is this correctly categorized?" title will come.
4. Rotate the screen, app will crash.
 
Screenshot_2019-11-13-17-43-59
Screenshot_2019-11-13-17-44-11

**Device and Android version:** 

SONY XPERIA Android version 5

 
**Commons app version:** 

2.11.0

**Screen-shots:** 

**Would you like to work on the issue?**

YES

### PR DESCRIPTION
**Pull request by DeepaMalani**
**Description (required)**

Fixes #3203 App Crash when rotating the Peer Review Screen

What changes did you make and why?

The app was getting crash for Review Screen because we were getting a null value on-device rotation for the getUser method.

I wrote code to save the value of the getUser before it gets destroyed when we rotate the phone.

**Tests performed (required)**

Manual

Tested {build variant, e.g. ProdDebug} on {name of device or emulator} with API level {API level}.

Sony Device API 21
Pixel XL API 28 Emulator
Nexus 6 API 26 

**Screenshots showing what changed (optional - for UI changes)**

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/review/ReviewImageFragment.java [hunk 1]
```diff
@@ -40,6 +42,9 @@
     @BindView(R.id.button_no)
     Button noButton;
 
+    private final String SAVED_USER = "saved_user";
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewImageFragment.java]: Could you please add a Javadoc comment explaining what the variable contains, with an example?

File: app/src/main/java/fr/free/nrw/commons/review/ReviewImageFragment.java [hunk 2]
```diff
@@ -40,6 +42,9 @@
...
+    private String user;
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewImageFragment.java]: Javadoc. Please make sure the difference with the `user` variable is easy to understand for any newcomer.

**Comment by DeepaMalani** [comment]
Hi Nicolas,
I have added java comments for the above code.

Thanks,
Deepa

File: app/src/main/java/fr/free/nrw/commons/review/ReviewImageFragment.java [hunk 3]
```diff
@@ -103,7 +111,15 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                 break;
             case THANKS:
                 question = getString(R.string.review_thanks);
-                explanation = getString(R.string.review_thanks_explanation, getReviewActivity().reviewController.firstRevision.getUser());
+
+                //Get existing user name if it is already saved using savedInstanceState else get from reviewController
+                if(savedInstanceState == null)
```
Comment (madhurgupta10) [inline app/src/main/java/fr/free/nrw/commons/review/ReviewImageFragment.java]: Could you please wrap the if-else statement in {}?

**Comment by DeepaMalani** [comment]
Hi Madhur,

I added curly braces for if-else statements.

**Comment by ashishkumar468** [comment]
LGTM :), Thanks @DeepaMalani. Happy to merge after @madhurgupta10 confirms that his comments are addressed.

**Comment by madhurgupta10** [comment]
> LGTM :), Thanks @DeepaMalani. Happy to merge after @madhurgupta10 confirms that his comments are addressed.

@DeepaMalani Good Work! 
@ashishkumar0207  You can merge it, no more changes needed from my side.

**Pull request merged**